src/System/Hapistrano.hs

```diff
@@ -330,6 +331,13 @@ restartServerCommand = do
     Nothing -> return "No command given for restart action."
     Just cmd -> runCommand (host conf) cmd
 
+cleanBuildScript :: [String] -> [String]
+cleanBuildScript allScriptLines = filter (not . isCommentOrEmpty) allScriptLines
+  where
+    isCommentOrEmpty line = isEmpty line || isComment line
```

Comment:
@jsl @juanpaucar What should happen if there's a leading space? I don't fully understand the original question.

As far as adding megaparsec as a dependency, it wouldn't be a big change in number of dependencies. Just about two additional things to install (at least with a very recent set of packages).